wayback/_client.py

```diff
@@ -540,7 +540,15 @@ def search(self, url, *, matchType=None, limit=None, offset=None,
                 else:
                     status_code = int(data.status_code)
                 length = None if data.length == '-' else int(data.length)
-                capture_time = _utils.parse_timestamp(data.timestamp)
+                # Fix bad timestamps
+                timestamp_chars = list(data.timestamp)
+                # If the timestamp has a day of "00"
+                if timestamp_chars[6:8] == ['0', '0']:
+                    timestamp_chars[7] = '1'
+                # If the timestamp has an hour of "24"
+                if timestamp_chars[8:10] == ['2', '4']:
+                    timestamp_chars[9] = '3'
+                capture_time = _utils.parse_timestamp("".join(timestamp_chars))
```

Comment:
This is complex enough that it’s probably better to put it in `_utils.parse_timestamp()` or in a helper function (e.g. `correct_invalid_timestamp()` or something).

Side note: single quotes on the `join()` call, please! :)

Reply:
Also, I *think* the right hour for this is probably `0`, not `23`. It reads to me like a rollover issue, where hour 24 should be the 0th hour of the next day. Does that seem right to you?

Reply:
Welp, there’s no clear answer on the right general solution here yet, but the IA folks dug under the hood and it seems like the right correction here is actually to *drop* the `00` part from the timestamp in this case, so we should be trying to parse `200008241623` (the underlying record’s raw timestamp is `2000080024162315` [which has too many digits] and the server-sent `Date` header is `Thu, 24 Aug 2000 16:21:19 GMT`, so it looks like some software error caused an extra `00` to be shoved in the middle). More info to come here as I receive it.

Reply:
There are 2 ways I could solve this, attempt to parse the string and upon failure apply the changes, then parse again in the utils function. Another way would be to detect the amount of digits and pre-emptively correct it, then parse it. Which would you prefer?

Reply:
> detect the amount of digits

Oh, sorry, I guess I was not clear. We can’t do this. The raw timestamp with too many digits isn’t available to us — we only know that was the case here because someone at the Internet Archive looked at the raw file that the CDX record was generated from. That raw data is not publicly available (or at least not easily).

> attempt to parse the string and upon failure apply the changes, then parse again in the utils function

I think what you’re doing right now to detect and solve the situation (looking for a `00` in the day-of-month part of the timestamp) is great! I just meant that you should put all that code into its own function, or put it in `_utils.parse_timestamp()` instead of in `WaybackClient.search()`.